ca/ecdsa_allowed_list.go

```diff
@@ -0,0 +1,72 @@
+package ca
```

Comment:
I think the interface here is kinda muddied. There are two different ways to make an allowlist (from config or from file), but they live in different source files (from config here, from file in ca.go). One of these mechanisms creates an allowlist from whole cloth, the other requires that an empty one already be created and then populates it. What happens if I create an allowlist from a config, and then *also* call `SetECDSAAllowedListFile` on it?

I think you recognized this code-smell yourself, since you commented on the `ECDSAAllowedList` field being exported even though it shouldn't be. But I think the answer is, rather than having these two very different ways to construct an allowlist, trying to make the two paths look as similar as possible.

I'd organize it like:
```go
// this is the same, but should have an additional field to store its own reloader
type allowList struct {...}
// this one replaces the current Set...File method: creates an allowList and starts and saves the reloader
func NewFromFile(filename string) *allowList {...}
// this one replaces LoadFromConfig, and sets the inner reloader to nil
func NewFromConfig(regIDs []int64) *allowList {...}
// this one is the same as now
func (l *allowList) load(contents []byte) error {...}
// this one is new, and calls .Stop() on the underlying reloader
func (l *allowList) Stop() error {...}
```

This way `main.go` can just can just construct an allowlist one way or the other, and then pass the resulting allowlist in to the CA constructor. It also means that the `ECDSAAllowedList` field on the CA struct doesn't have to be exported, because that field will be set by the constructor. And finally `CertificateAuthorityImpl.Stop()` can call its `allowList.Stop()` during clean shutdown.

Reply:
This is fantastic and seems absolutely obvious to now, thank you!